src/asset/MountArchive.cpp

```diff
@@ -0,0 +1,71 @@
+#include "MountArchive.h"
+
+#include "bit7z/bitarchivereader.hpp"
+#include "utils/Log.h"
+
+MountArchive::MountArchive(const std::filesystem::path& archive_path)
+    : Mount(archive_path), reader(std::make_unique<bit7z::BitArchiveReader>(library, path.string())) {
+}
+
+MountArchive::~MountArchive() {
+}
+
+void MountArchive::load() {
+    reset_reader();
+
+    try {
+        reader = std::make_unique<bit7z::BitArchiveReader>(library, get_path().string());
+        reader->test();
+    }
+    catch (const std::exception& exception) {
+        throw exception;
+    }
+
+    load_cache();
+}
+
+bool MountArchive::seek_file(const std::string& path) {
+    return static_cache.contains(path);
+}
+
+std::vector<uint8_t> MountArchive::fetch_data(const std::string& path) {
+    uint32_t index = static_cache.at(path);
+    std::vector<uint8_t> buffer;
+
+    try {
+        reader->extractTo(buffer, index);
+    }
+    catch (const std::exception& exception) {
+        throw exception;
+    }
+
+    return buffer;
+}
+
+void MountArchive::load_cache() {
+    const std::filesystem::path cache_file_path = get_path().replace_filename("cache");
+
+    if (std::filesystem::exists(cache_file_path)) {
+        return;
+    }
+
+    const auto items = reader->items();
+    for (const auto& item : items) {
+        static_cache.at(item.path().replace(0, 2, "/")) = item.index();

```

Comment:
[2:54 PM]scatterflower: I mentioned this in a comment but it wasn't changed:
static_cache.at(item.path().replace(0, 2, "/")) = item.index();
[2:54 PM]scatterflower: is the point of this to strip the drive letter? won't that break on cross-platform?
[2:55 PM]Salanto: 7z returns files formatted as :

"\\characters\phoenix\char.ini" 
[2:55 PM]Salanto: I strip away the prefix.
[2:55 PM]scatterflower: ah okay
[2:55 PM]scatterflower: might wanna put that in a comment